packages/babel-plugin/src/utils/object-property-to-string.ts

```diff
@@ -0,0 +1,88 @@
+import * as t from '@babel/types';
+
+import type { Metadata } from '../types';
+
+import { evaluateExpression } from './evaluate-expression';
+
+type ExpressionToString = (expression: t.Expression, meta: Metadata) => string;
+
+const templateLiteralToString = (
+  template: t.TemplateLiteral,
+  meta: Metadata,
+  expressionToString: ExpressionToString
+) => {
+  let result = '';
+
+  for (let i = 0; i < template.quasis.length; i += 1) {
+    result += template.quasis[i].value.raw;
+
+    if (i < template.expressions.length) {
+      const expression = template.expressions[i];
+      if (t.isTSType(expression)) {
+        // Passed a type instead of a value
+        // e.g. `${any}`
+        throw new Error(`${template.type} has a type instead of a value`);
+      }
+      const evaluatedExpression = evaluateExpression(expression, meta);
+      result += expressionToString(evaluatedExpression.value, evaluatedExpression.meta);
+    }
+  }
+
+  return result;
+};
+
+const binaryExpressionToString = (
+  expression: t.BinaryExpression,
+  meta: Metadata,
+  expressionToString: ExpressionToString
+): string => {
+  const { left, right, operator } = expression;
+
+  if (operator === '+' && t.isExpression(left)) {
+    const leftValue = expressionToString(left, meta);
+    const rightValue = expressionToString(right, meta);
+
+    return `${leftValue}${rightValue}`;
+  }
+
+  throw new Error(`Cannot use ${operator} for string operation. Use + for string concatenation`);
+};
+
+const expressionToString: ExpressionToString = (expression, meta) => {
+  // handles {'key-name': 'value'} or {1: 'value'}
```

Comment:
Not sure if we should also handle `isNumericLiteral`. 
I can't think of any css property defined with a number, or any scenario where a selector could be a number. Keeping it shouldn't cause any issue though.

@JakeLane, @liamqma  any thoughts on this?

Reply:
Yeah true, when I was writing it I was thinking about all Objects in general 😅 Maybe number should be unsupported so it can throw an error?

Reply:
On second thought, I think we do need to keep the numeric evaluation, otherwise we can do things like:
`['field' + 1]: { ... }`

Reply:
I thought it was checking for numbers only. Yeahh, we should keep it if it deals with alphanumerics :)